src/language-js/comments.js

```diff
@@ -930,7 +933,7 @@ function isBlockComment(comment) {
   return comment.type === "Block" || comment.type === "CommentBlock";
 }
 
-function hasLeadingComment(node, fn = () => true) {
+function hasLeadingComment(node, fn = (_) => true) {
```

Comment:
Can this done in other way?

Reply:
Yes, good catch.

Here is one way that works for me:

```suggestion
/**
 * @param {Object} node
 * @param {(comment: Object) => boolean} fn
 * @returns boolean
 */
function hasLeadingComment(node, fn = () => true) {
```

In this case, I think we should add a TODO item to use more specific types for the node parameter and comment function argument.

But I am now thinking we can just drop the `fn` argument since `git grep -w hasLeadingComments` shows it never used.

Reply:
It isn't practical to add a TODO in every place where better types are needed. Please don't do that. The type `any` ([AKA `Object`](https://github.com/microsoft/TypeScript/issues/20337)) is already a TODO in itself. You can also use the `?` type (another synonym for `any`, `*` works too) to make it even more explicit if you want: `/** @param {?} node */`.

Reply:
I don't know anything about  this, just  wonder if default value of argument function with  5 parameters is  `()=> true`, we  won't list all the  5 parameter, right ?

Reply:
I think `any` would be easiest to search for. Any other ideas?

> if default value of argument function with 5 parameters is `()=> true`

I am not sure if I understand the question right. It does work for me if I would only keep the `@param` comment for the `fn` argument. But I think this would look like sloppy, unfinished work.

As a side point, I think it should not take too much work to typecheck most of the other sources once this update is finished. Then I think it would be nice to start typecheck with `"strict": true` on as many sources as possible, which I think would imply defining all arguments in the JSDoc comments.

Reply:
> we won't list all the 5 parameter, right ?

We will. Adding types mostly means adding types to function signatures.

Reply:
You are saying we  might have code like

```js
function hasLeadingComment(node, fn = (_1, _2, _3, _4,_5) => true)
```

?

Reply:
If we  have

```js
function foo(node, fn = returnTure) {}

function bar(node, fn = returnTure) {}
```

and these  two  `fn` takes different parameters, can  it pass types test ?

Reply:
"adding types to function signatures" - I meant the JSDoc comments, not `_1, _2, ...`.

`() => true` *is* assignable to e.g. `(param1: number) => boolean`.

Reply:
Thank you!